crypto/rand/rand_lib.c

```diff
@@ -434,8 +444,13 @@ void rand_pool_free(RAND_POOL *pool)
      * to rand_pool_attach() as `const unsigned char*`.
      * (see corresponding comment in rand_pool_attach()).
      */
-    if (!pool->attached)
-        OPENSSL_secure_clear_free(pool->buffer, pool->max_len);
+    if (!pool->attached) {
+        if (pool->secure)
+            OPENSSL_secure_clear_free(pool->buffer, pool->max_len);
+        else
+            OPENSSL_clear_free(pool->buffer, pool->max_len);
```

Comment:
I could probably use pool->len instead of max_len in this non-secure case.
At least a small performance improvement, when I can't find a way how to work
around the sanitizer.

Reply:
So I leave the possible performance improvement to @paulidale here as well.